### PR TITLE
Add type checking in `Object.perform`

### PR DESCRIPTION
**Pull request by brianlheim**
In response to #2615.

Attempting to interpret `3.perform([0])` now results in:
```
ERROR: First element of array must be a Symbol selector.
   Integer 0
ERROR: Primitive '_ObjectPerform' failed.
Wrong type.
```
and `3.perform(['+', 5])` still gives `8`.

Todo:
- [x] Document the behavior of `Object.perform(Array | List)`

### CODE REVIEW

File: lang/LangPrimSource/PyrPrimitive.cpp [hunk 1]
```diff
@@ -1493,6 +1493,11 @@ int objectPerform(struct VMGlobals *g, int numArgsPushed)
 			return errFailed;
 		}
 		selSlot = array->slots;
+        if (!IsSym(selSlot)) {
```
Comment (timblechmann) [inline lang/LangPrimSource/PyrPrimitive.cpp]: there are probably other instances in similar code

**Comment by snappizz** [comment]
If we document using an Array as a selector, it should be discouraged. We already have Object:performMsg, and `x.perform([y, z])` just looks like a mistake.

**Comment by brianlheim** [comment]
Hmm. Good point. If really necessary, the call can be written as `x.perform(*arr)`, but it seems awkward and style-inconsistent that an array as the first argument can potentially be expanded to fill the second argument. Since this code seems to be pretty old, though, I was worried about suddenly removing that behavior. If it's not documented, there's not really anything to deprecate, so could it be removed right now?

**Comment by snappizz** [comment]
I suggest documenting the quirk and discouraging it, and suggesting that Object:performMsg be used instead.

**Pull request merged**